core/services/ardupilot_manager/mavlink_proxy/MAVLinkRouter.py

```diff
@@ -26,9 +26,9 @@ def assemble_command(self, master_endpoint: Endpoint) -> str:
         # Convert endpoint format to mavlink-router format
         def convert_endpoint(endpoint: Endpoint) -> str:
             # TCP uses a special argument and only works with localhost
-            if endpoint.connection_type == EndpointType.TCPServer:
-                return f"--tcp-port {endpoint.argument}"
             if endpoint.connection_type == EndpointType.TCPClient:
+                return f"--tcp-port {endpoint.argument}"
```

Comment:
Wait, here you are using tcp-port as a client and in line 68 as a server.

IIRC the tcp-port receives only the port because it binds on the host computer, being a server by nature.

Reply:
good catch, it turned out that only the master endpoint was switched